cmake/modules/AddSwift.cmake

```diff
@@ -707,6 +715,14 @@ function(_add_swift_library_single target name)
       add_dependencies(${target}_IMPLIB ${${target}_IMPORT_LIBRARY})
     endif()
     set_property(TARGET "${target}" PROPERTY NO_SONAME ON)
+  elseif("${SWIFTLIB_SINGLE_SDK}" STREQUAL "ANDROID")
```

Comment:
This code path is pretty cheesy. Do you think you can refactor it out in some way?